galley/pkg/config/analysis/analyzers/authz/authorizationpolicies.go

```diff
@@ -0,0 +1,115 @@
+
+	"istio.io/api/security/v1beta1"
+
+	"istio.io/istio/galley/pkg/config/analysis"
+	"istio.io/istio/pkg/config/resource"
+	"istio.io/istio/pkg/config/schema/collection"
+	"istio.io/istio/pkg/config/schema/collections"
+)
+
+// AuthorizationPoliciesAnalyzer checks the validity of authorization policies
+type AuthorizationPoliciesAnalyzer struct{}
+
+var _ analysis.Analyzer = &AuthorizationPoliciesAnalyzer{}
+
+func (a *AuthorizationPoliciesAnalyzer) Metadata() analysis.Metadata {
+	return analysis.Metadata{
+		Name:        "auth.AuthorizationPoliciesAnalyzer",
+		Description: "Checks the validity of authorization policies",
+		Inputs: collection.Names{
+			collections.IstioSecurityV1Beta1Authorizationpolicies.Name(),
+			collections.K8SAppsV1Deployments.Name(),
+			collections.K8SCoreV1Namespaces.Name(),
+			collections.IstioNetworkingV1Alpha3Serviceentries.Name(),
+			collections.K8SCoreV1Services.Name(),
+		},
+	}
+}
+
+func (a *AuthorizationPoliciesAnalyzer) Analyze(c analysis.Context) {
+	serviceEntryHosts := util.InitServiceEntryHostMap(c)
+
+	c.ForEach(collections.IstioSecurityV1Beta1Authorizationpolicies.Name(), func(r *resource.Instance) bool {
+		a.analyzeNoMatchingWorkloads(r, c)
+		a.analyzeNamespaceNotFound(r, c)
+		a.analyzeHostNotFound(r, c, serviceEntryHosts)
+		return true
+	})
+}
+
+func (a *AuthorizationPoliciesAnalyzer) analyzeNoMatchingWorkloads(r *resource.Instance, c analysis.Context) {
+	ap := r.Message.(*v1beta1.AuthorizationPolicy)
+	ns := r.Metadata.FullName.Namespace
+	apSelector := k8s_labels.SelectorFromSet(ap.Selector.MatchLabels)
+
+	hasMatchingPods := false
+
+	c.ForEach(collections.K8SAppsV1Deployments.Name(), func(s *resource.Instance) bool {
+		d := s.Message.(*apps_v1.Deployment)
+		pLabels := k8s_labels.Set(d.Spec.Template.Labels)
+
+		if d.Namespace == ns.String() && apSelector.Matches(pLabels) {
+			hasMatchingPods = true
+			return false
+		}
+
+		return true
+	})
+
+	if !hasMatchingPods {
+		c.Report(collections.IstioSecurityV1Beta1Authorizationpolicies.Name(), msg.NewNoMatchingWorkloadsFound(r, apSelector.String()))
+	}
+}
+
+func (a *AuthorizationPoliciesAnalyzer) analyzeNamespaceNotFound(r *resource.Instance, c analysis.Context) {
+	ap := r.Message.(*v1beta1.AuthorizationPolicy)
+
+	for _, rule := range ap.Rules {
+		for _, from := range rule.From {
+			for _, ns := range from.Source.Namespaces {
+				if !c.Exists(collections.K8SCoreV1Namespaces.Name(), resource.NewFullName("", resource.LocalName(ns))) {
+					c.Report(collections.IstioSecurityV1Beta1Authorizationpolicies.Name(), msg.NewReferencedResourceNotFound(r, "namespace", ns))
+				}
+			}
+		}
+	}
+}
+
+func (a *AuthorizationPoliciesAnalyzer) analyzeHostNotFound(r *resource.Instance, c analysis.Context,
+	serviceEntryHosts map[util.ScopedFqdn]*v1alpha3.ServiceEntry) {
+	ap := r.Message.(*v1beta1.AuthorizationPolicy)
+	for _, rule := range ap.Rules {
+		for _, to := range rule.To {
+			for _, host := range to.Operation.Hosts {
```

Comment:
same as namespace, the host supports prefix/suffix/existence match.

Reply:
Do you mind setting some examples here? The suffix/prefix/existence match is only applied to the service or can it be applied to other places in the FQDN?
`*-xyz.namespace.cluster.local`, `xyz-*.namespace.cluster.local` and `*.namespace.cluster.local` are correct values, right?